src/Symfony/Component/TypeInfo/Type/CollectionType.php

```diff
@@ -92,6 +92,39 @@ public function wrappedTypeIsSatisfiedBy(callable $specification): bool
         return $this->getWrappedType()->isSatisfiedBy($specification);
     }
 
+    public function accepts(mixed $value): bool
+    {
+        if (!parent::accepts($value)) {
+            return false;
+        }
+
+        // array is not a list
+        if ($this->isList() && \is_array($value) && !array_is_list($value)) {
+            return false;
+        }
+
+        $keyType = $this->getCollectionKeyType();
+        $valueType = $this->getCollectionValueType();
+
+        $i = 0;
+
+        foreach ($value as $k => $v) {
```

Comment:
The value will be iterated. Is it possible to analyze whether `$value` is an instance of a class with generic type declaration and check type compatibility?

```php
/** @template-implements ArrayAccess<string, string> */
class StringCollection extends ArrayAccess
{
    public function offsetGet(string $key): string 
    { /* … */ }
}
```

Reply:
Unfortunately, I have dig and I wasn't able to find any way to get all keys/values of an `ArrayAccess` class (but maybe you have a way in mind?).

This however leads us to a question. How do we behave in that case? Should we accept or reject the value?